api/routes/settings.ts

```diff
@@ -0,0 +1,63 @@
+import type { FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { Type } from '@sinclair/typebox';
+
+export const settingRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.put(
+    '/update-my-profileui',
+    {
+      schema: {
+        body: Type.Object({
+          profileUI: Type.Object({
+            isLocked: Type.Boolean(),
+            showAbout: Type.Boolean(),
+            showCerts: Type.Boolean(),
+            showDonation: Type.Boolean(),
+            showHeatMap: Type.Boolean(),
+            showLocation: Type.Boolean(),
+            showName: Type.Boolean(),
+            showPoints: Type.Boolean(),
+            showPortfolio: Type.Boolean(),
+            showTimeLine: Type.Boolean()
+          })
+        }),
+        200: Type.Object({
+          message: Type.String(),
+          type: Type.Literal('success')
+        }),
+        500: Type.Object({
+          message: Type.String(),
+          type: Type.Literal('danger')
+        })
+      }
+    },
+    async req => {
+      try {
+        await fastify.prisma.user.update({
+          where: { id: req.session.user.id },
+          data: {
+            profileUI: {
+              isLocked: req.body.profileUI.isLocked,
+              showAbout: req.body.profileUI.showAbout,
+              showCerts: req.body.profileUI.showCerts,
+              showDonation: req.body.profileUI.showDonation,
+              showHeatMap: req.body.profileUI.showHeatMap,
+              showLocation: req.body.profileUI.showLocation,
+              showName: req.body.profileUI.showName,
+              showPoints: req.body.profileUI.showPoints,
+              showPortfolio: req.body.profileUI.showPortfolio,
+              showTimeLine: req.body.profileUI.showTimeLine
+            }
```

Comment:
**General Question**

I realise we now have the schema - `req.body.profileUI` content should be validated as only `true | false` - but is there still the possibility for a Camper to create a malformed request to update their user information:

```javascript
fetch('/update-my-profileui`, { body: { profileUI: { isLocked: "bad data" } } })
```

Just double-checking, because Mongodb will be happy with whatever we give it.

Reply:
That's correct. Also Prisma offers further protection, since it rejects non-boolean values, too.

You can test this by submitting bad requests and seeing what response you get.